feat: add `math/base/assert/is-probabilityf`

vivekmaurya001 opened the pull request:
Progresses None

## Description

- adds math/base/assert/is-probabilityf, which would be the single-precision equivalent for [math/base/assert/is-probability](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/assert/is-probability).

This pull request:


## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

 - [x] Read, understood, and followed the [contributing guidelines](https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md)


* * *

@stdlib-js/reviewers 

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

vivekmaurya001 committed:
add math/base/assert/is-probabilityf

vivekmaurya001 committed:
Merge branch 'is_probabilityf' of https://github.com/vivekmaurya001/stdlib into is_probabilityf

vivekmaurya001 committed:
add math/base/assert/is-probabilityf

vivekmaurya001 committed:
add math/base/assert/is-probabilityf

vivekmaurya001 commented:
/stdlib update-copyright-years

stdlib-bot committed:
chore: update copyright years

vivekmaurya001 committed:
add math/base/assert/is-probabilityf

vivekmaurya001 committed:
Merge branch 'is_probabilityf' of https://github.com/vivekmaurya001/stdlib into is_probabilityf

vivekmaurya001 committed:
add math/base/assert/is-probabilityf

vivekmaurya001 committed:
add math/base/assert/is-probabilityf

vivekmaurya001 committed:
add math/base/assert/is-probabilityf

vivekmaurya001 commented:
@kgryte , please give some reveiw

vivekmaurya001 committed:
Merge branch 'stdlib-js:develop' into is_probabilityf

vivekmaurya001 committed:
Merge branch 'stdlib-js:develop' into is_probabilityf

vivekmaurya001 commented:
/stdlib update-copyright-years

stdlib-bot committed:
chore: update copyright years

gunjjoshi commented on lib/node_modules/@stdlib/math/base/assert/is-probabilityf/README.md:
````diff
@@ -0,0 +1,182 @@
+<!-- eslint no-undef: "error" -->
+
+```javascript
+var randu = require( '@stdlib/random/base/randu' );
+var isProbabilityf = require( '@stdlib/math/base/assert/is-probabilityf' );
+
+var bool;
+var x;
+var i;
+
+for ( i = 0; i < 100; i++ ) {
+    x = ( randu()*2.0 ) - 1.0;
+    bool = isProbabilityf( x );
````
@vivekmaurya001 We can use `@stdlib/random/array/uniform` here, and initialize `x` as an array.

gunjjoshi replied:
This can be done in `examples/index.js` too.

vivekmaurya001 replied:
done the changes ! please take a look

gunjjoshi commented on lib/node_modules/@stdlib/math/base/assert/is-probabilityf/README.md:
````diff
@@ -0,0 +1,182 @@
+Tests if a single-precision floating-point number is a probability.
+
+```c
+bool out = stdlib_base_is_probabilityf( 0.5 );
````
```suggestion
bool out = stdlib_base_is_probabilityf( 0.5f );
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/assert/is-probabilityf/README.md:
```diff
@@ -0,0 +1,182 @@
+bool out = stdlib_base_is_probabilityf( 0.5 );
+// returns true
+
+out = stdlib_base_is_probabilityf( 3.14 );
```
```suggestion
out = stdlib_base_is_probabilityf( 3.14f );
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/assert/is-probabilityf/benchmark/benchmark.js:
```diff
@@ -0,0 +1,58 @@
+
+var bench = require( '@stdlib/bench' );
+var Float32Array = require( '@stdlib/array/float32' );
+var randu = require( '@stdlib/random/base/randu' );
```
@vivekmaurya001 Can you please change this and the code below accordingly, to use `@stdlib/random/array/uniform`?

gunjjoshi replied:
Same can be done for `benchmark.native.js` as well.

vivekmaurya001 replied:
done

gunjjoshi commented on lib/node_modules/@stdlib/math/base/assert/is-probabilityf/benchmark/c/native/benchmark.c:
```diff
@@ -0,0 +1,137 @@
+
+	t = tic();
+	for ( i = 0; i < ITERATIONS; i++ ) {
+		b = stdlib_base_is_probabilityf( x[ i%100 ] );
```
```suggestion
		b = stdlib_base_is_probabilityf( x[ i % 100 ] );
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/assert/is-probabilityf/package.json:
```diff
@@ -0,0 +1,81 @@
+    "math",
+    "mathematics",
+    "float",
+    "double",
+    "dbl",
```
```suggestion
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/assert/is-probabilityf/src/main.c:
```diff
@@ -0,0 +1,39 @@
+* @example
+* #include <stdbool.h>
+*
+* bool out = stdlib_base_is_probabilityf( 0.5 );
```
```suggestion
* bool out = stdlib_base_is_probabilityf( 0.5f );
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/assert/is-probabilityf/src/main.c:
```diff
@@ -0,0 +1,39 @@
+* bool out = stdlib_base_is_probabilityf( 0.5 );
+* // returns true
+*
+* out = stdlib_base_is_probabilityf( 1.5 );
```
```suggestion
* out = stdlib_base_is_probabilityf( 1.5f );
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/assert/is-probabilityf/src/main.c:
```diff
@@ -0,0 +1,39 @@
+* // returns false
+*/
+bool stdlib_base_is_probabilityf( const float x ) {
+	return ( x >= 0.0 && x <= 1.0 );
```
```suggestion
	return ( x >= 0.0f && x <= 1.0f );
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/assert/is-probabilityf/test/test.js:
```diff
@@ -0,0 +1,99 @@
+	var x;
+	var i;
+	for ( i = 0; i < 1000; i++ ) {
+		x = -1.0 - ( randu()*1000.0 );
```
```suggestion
		x = -1.0 - ( randu() * 1000.0 );
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/assert/is-probabilityf/test/test.js:
```diff
@@ -0,0 +1,99 @@
+	var x;
+	var i;
+	for ( i = 0; i < 1000; i++ ) {
+		x = ( randu()*1000.0 ) + 2.0;
```
```suggestion
		x = ( randu() * 1000.0 ) + 2.0;
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/assert/is-probabilityf/test/test.js:
```diff
@@ -0,0 +1,99 @@
+
+tape( 'the function returns `false` if provided `NaN`', function test( t ) {
+	t.equal( isProbabilityf( NaN ), false, 'returns false' );
+	t.equal( isProbabilityf( 0.0/0.0 ), false, 'returns false' );
```
```suggestion
	t.equal( isProbabilityf( 0.0 / 0.0 ), false, 'returns false' );
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/assert/is-probabilityf/test/test.native.js:
```diff
@@ -0,0 +1,108 @@
+	var x;
+	var i;
+	for ( i = 0; i < 1000; i++ ) {
+		x = -1.0 - ( randu()*1000.0 );
```
```suggestion
		x = -1.0 - ( randu() * 1000.0 );
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/assert/is-probabilityf/test/test.native.js:
```diff
@@ -0,0 +1,108 @@
+	var x;
+	var i;
+	for ( i = 0; i < 1000; i++ ) {
+		x = ( randu()*1000.0 ) + 2.0;
```
```suggestion
		x = ( randu() * 1000.0 ) + 2.0;
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/assert/is-probabilityf/test/test.native.js:
```diff
@@ -0,0 +1,108 @@
+
+tape( 'the function returns `false` if provided `NaN`', opts, function test( t ) {
+	t.equal( isProbabilityf( NaN ), false, 'returns false' );
+	t.equal( isProbabilityf( 0.0/0.0 ), false, 'returns false' );
```
```suggestion
	t.equal( isProbabilityf( 0.0 / 0.0 ), false, 'returns false' );
```

gunjjoshi committed:
Apply suggestions from code review

Signed-off-by: Gunj Joshi <gunjjoshi8372@gmail.com>

gunjjoshi commented on lib/node_modules/@stdlib/math/base/assert/is-probabilityf/test/test.js:
```diff
@@ -0,0 +1,99 @@
+	for ( i = 0; i < 1000; i++ ) {
+		x = -1.0 - ( randu() * 1000.0 );
+		bool = isProbabilityf( x );
+		t.equal( bool, false, 'returns false when provided '+x );
```
```suggestion
		t.equal( bool, false, 'returns expected value when provided '+x );
```

gunjjoshi replied:
@vivekmaurya001 Can you please change this everywhere else in this file and in `test.native.js`?

vivekmaurya001 replied:
should i replace the return true also ? , for now i have done with false

gunjjoshi replied:
Yes, @vivekmaurya001 .

vivekmaurya001 replied:
ook

vivekmaurya001 replied:
done

vivekmaurya001 committed:
Merge branch 'stdlib-js:develop' into is_probabilityf

vivekmaurya001 committed:
add math/base/assert/is-probabilityf

vivekmaurya001 committed:
add math/base/assert/is-probabilityf

vivekmaurya001 committed:
add math/base/assert/is-probabilityf

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: passed
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: passed
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: passed
---

vivekmaurya001 committed:
add math/base/assert/is-probabilityf

gunjjoshi commented on lib/node_modules/@stdlib/math/base/assert/is-probabilityf/test/test.js:
```diff
@@ -0,0 +1,99 @@
+
+tape( 'the function returns `false` if provided `NaN`', function test( t ) {
+	t.equal( isProbabilityf( NaN ), false, 'returns expected value' );
+	t.equal( isProbabilityf( 0.0/0.0 ), false, 'returns expected value' );
```
```suggestion
	t.equal( isProbabilityf( 0.0 / 0.0 ), false, 'returns expected value' );
```

gunjjoshi committed:
Update lib/node_modules/@stdlib/math/base/assert/is-probabilityf/test/test.js

Signed-off-by: Gunj Joshi <gunjjoshi8372@gmail.com>

vivekmaurya001 commented:
Good morning , @kgryte  please take a look
